index.js

```diff
@@ -13,9 +10,17 @@ ajv.addKeyword('separator', {
   valid: true,
   errors: false,
   compile: (schema) => (data, dataPath, parentData, parentDataProperty) => {
-    parentData[parentDataProperty] = data === '' ? [] : data.split(schema)
+    if (parentData && parentDataProperty) {
+      parentData[parentDataProperty] = data === '' ? [] : data.split(schema)
+    } else {
+      const {
+        parentData: pData,
+        parentDataProperty: pDataProperty
+      } = dataPath
+      pData[pDataProperty] = data === '' ? [] : data.split(schema)
+    }
   }
```

Comment:
can you explain what this does and why it's needed?

Reply:
In some cases, `parentData` and `parentDataProperty` will be undefined. We need to fall back to the `dataPath` object to provide those values. I have added a comment to the above.

Reply:
I'm curious how you found this out. Are there tests failing if you don't have this fallback in place? And do you know exactly when it happens?

Reply:
Without the fall back there are three failing tests. I added a console logger to inspect the properties being passed to the separator compile function and noted that the parameters used were undefined. 

I added a conditional check to make sure the parent attributes were defined but this applied the separator logic incorrectly and still left failing tests. Attached below is one such test. I then went further and used the properties from the `dataPath` object to replace the undefined values. The tests then passed as expected. 

![image](https://user-images.githubusercontent.com/5331111/109839732-51ace000-7c3f-11eb-80fe-aae70cd621de.png)